src/main/java/stsjorbsmod/actions/DamageWithOnKillEffectAction.java

```diff
@@ -0,0 +1,43 @@
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.powers.MinionPower;
+
+public class DamageWithOnKillEffectAction extends AbstractGameAction {
+    private DamageInfo info;
+    private Runnable onKillCallback;
+    private boolean worksOnMinions;
+
+
+    public DamageWithOnKillEffectAction(AbstractCreature target, DamageInfo info, Runnable onKillCallback) {
+        this.info = info;
+        this.setValues(target, info);
+        this.actionType = ActionType.DAMAGE;
+        this.duration = 0.1F;
+        this.onKillCallback = onKillCallback;
+        this.worksOnMinions = false;
+    }
+
+    public DamageWithOnKillEffectAction(AbstractCreature target, DamageInfo info, Runnable onKillCallback, boolean worksOnMinions) {
+        this.info = info;
+        this.setValues(target, info);
+        this.actionType = ActionType.DAMAGE;
+        this.duration = 0.1F;
+        this.onKillCallback = onKillCallback;
+        this.worksOnMinions = worksOnMinions;
+    }
+
+    public void update() {
+        if (this.duration == 0.1F && this.target != null) {
+            this.target.damage(this.info);
```

Comment:
This still needs to have a clearPostCombatActions check; I would recommend modelling this update method very closely after what you did for `HealOnFatalDamageAction`, I think it's exactly right if you swap out its HealAction part for the new runnable thing